internal/grpc/services/gateway/usershareprovider.go

```diff
@@ -170,6 +170,7 @@ func (s *svc) ListReceivedShares(ctx context.Context, req *collaboration.ListRec
 	// The `ListStorageSpaces` method in sharesstorageprovider/sharesstorageprovider.go needs the etags.
 	shareMetaData := make(map[string]share.Metadata, len(res.Shares))
 	for _, rs := range res.Shares {
+		// Stat .. here, only for etag and mtime? and then we throw the rest away?
```

Comment:
absolutely. The usershareprovider should not stat resources. if you need the resource info a client has to stat the resource yourself or use a cached version... here a client would also be the sharesstorageprovider. When listing spaces it may only need etag and mtime, but when listingf the share jail it needs the complete resource info. something for a subsequent pr.

Reply:
OK. Got it. Could you express this in the comment? Like 
```suggestion
		// FIXME: usershareprovider should not stat resources
```